config.go

```diff
@@ -0,0 +1,39 @@
+package timescaledb
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+type config struct {
+	// Connection URL in the form specified in the libpq docs,
+	// see https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING):
+	// postgresql://[user[:password]@][netloc][:port][,...][/dbname][?param1=value1&...]
+	URL              string
+	PushInterval     time.Duration
+	ConcurrentWrites int
+}
+
+func getEnvConfig(env map[string]string) (config, error) {
+	url, ok := env["K6_TIMESCALEDB_URL"]
+	if !ok {
+		return config{}, fmt.Errorf("invalid K6_TIMESCALEDB_URL")
+	}
+
+	pushInterval, err := time.ParseDuration(env["K6_TIMESCALEDB_PUSH_INTERVAL"])
+	if err != nil {
+		return config{}, fmt.Errorf("invalid K6_TIMESCALEDB_PUSH_INTERVAL: %w", err)
+	}
+
+	concurrentWrites, err := strconv.Atoi(env["K6_TIMESCALEDB_CONCURRENT_WRITES"])
+	if err != nil {
+		return config{}, fmt.Errorf("invalid K6_TIMESCALEDB_CONCURRENT_WRITES: %w", err)
+	}
+
+	return config{
+		URL:              url,
+		PushInterval:     pushInterval,
+		ConcurrentWrites: concurrentWrites,
```

Comment:
You currently don't use concurrent writes anywhere and the previous way it worked is IMO ... wrong as it can race on the Thresholds update, so I am somewhat thinking it might be better if you drop it.

 But maybe first let's get to the part where we test the performance and figuring out if it will help ;)